### PR TITLE
Bot detector frontend_data log

### PR DESCRIPTION
**Pull request by alexandergull**
https://doboard.com/1/task/14884

### CODE REVIEW

**Review by svfcode — APPROVED** [review]

File: inc/cleantalk-common.php [hunk 1]
```diff
@@ -613,8 +613,10 @@ function apbct_get_sender_info()
             ? json_encode(Cookie::get('ct_has_input_focused'))
             : null,
         'cache_plugins_detected' => $cache_plugins_detected,
+        //bot detector data
         'bot_detector_fired_form_exclusions' => apbct__bot_detector_get_fired_exclusions(),
         'bot_detector_prepared_form_exclusions' => apbct__bot_detector_get_prepared_exclusion(),
+        'bot_detector_frontend_data_log' => apbct__bot_detector_get_fd_log(),
```
Comment (Glomberg) [inline inc/cleantalk-common.php:619]: No need to collect `apbct__bot_detector_get_fd_log` if the option `Use Anti-Spam by CleanTalk JavaScript library` is disabled.

File: js/src/apbct-public--2--public.js [hunk 2]
```diff
@@ -30,6 +31,7 @@ const ctFunctionFirstKey = function output(event) {
 
 // run cron jobs
 cronFormsHandler(2000);
+sendBotDetectorLogToAltSessions(500);
```
Comment (Glomberg) [inline js/src/apbct-public--2--public.js:34]: No need to send this ajax requests if the option `Use Anti-Spam by CleanTalk JavaScript library` is disabled.

**Review by Glomberg — CHANGES_REQUESTED** [review]

File: js/src/apbct-public--2--public.js (hunk 2 above)
Comment (alexandergull) [reply js/src/apbct-public--2--public.js:34]: 👍

**Review by AntonV1211 — APPROVED** [review]

File: js/src/apbct-public--2--public.js (hunk 2 above)
Comment (alexandergull) [reply js/src/apbct-public--2--public.js]: https://github.com/CleanTalk/wordpress-antispam/pull/509/commits/745a32454619fe68606cba10c4fff5704d25d508

File: inc/cleantalk-common.php (hunk 1 above)
Comment (alexandergull) [reply inc/cleantalk-common.php:619]: https://github.com/CleanTalk/wordpress-antispam/pull/509/commits/745a32454619fe68606cba10c4fff5704d25d508

File: tests/StandaloneFunctions/TestBotDetectorLogGathering.php
Comment (Glomberg) [inline tests/StandaloneFunctions/TestBotDetectorLogGathering.php:1]: 👍

**Review by Glomberg — APPROVED** [review]

**Review by Glomberg — APPROVED** [review]

**Pull request merged**